### ISSUE TITLE
Airdropping to an ENS address doesn't work

### ISSUE THREAD

**Issue by jrainville** [status-im/status-desktop#13780]
# Bug Report

## Description

We tried to send a community token to an ENS address and it didn't work.

## Steps to reproduce

1. Have a community minted token
2. Go to airdrop page and select that token
3. Select ethereum address and enter an ENS address (eg sirotin.eth)
4. Doesn't work

#### Expected behavior

Sending to an ENS address should be possible as it points to an ethereum address

#### Actual behavior

Error in the airdrop page

### Additional Information

- Status desktop version: 2.27RC5

**Comment by kounkou** [comment]
Hi @jrainville 

by `4. Doesn't work` do we mean that the address is considered invalid like in the below screenshot

![Image](https://github.com/status-im/status-desktop/assets/2589171/925ddaec-fb1e-47f0-b660-fa54f4b5d173)

**Comment by jrainville** [comment]
> by 4. Doesn't work do we mean that the address is considered invalid like in the below screenshot

@kounkou yes. It should accept ENS addresses and when sending, convert the ENS address to an ETH address to send.

It's possible however that we never supported it. If you check the code and see that we have no mention of ENS anywhere, it's worth it to postpone the implementation to later.

**Comment by kounkou** [comment]
I can see enough reference to converting Ethereum Name Service (ENS) names to ETH, fix should be very fast : 

https://github.com/status-im/status-desktop/blob/0146db8da8c355e20933abdbbd11cda7423d5bd0/src/app/modules/main/module.nim#L1146

at a lower level

https://github.com/status-im/status-go/blob/2fb6d615fd39ce8500ace8c26332f7ca3c127276/vendor/github.com/wealdtech/go-ens/v3/resolver.go#L186

**Comment by jrainville** [comment]
Right, I know we have some functions to resolve ENS, but I was wondering if that feature of sending airdrops ever supported it. If it never did, then it's not worth implementing it right now.

I created this issue because we assumed that it was working at some point and supported.

Maybe we were wrong, and in that case, this would be a feature and we can push it to a later milestone (we plan on revamping ENS support later)

**Comment by kounkou** [comment]
Sure, makes sense, thank you for clarifying. 
So I am just updating the validation function and performing some tests for now. I'll come back with results.

This modification assumes that an ENS name is of the form `*.eth` where is * is any strings.
I wonder if we would like to add a limit to the length though. I'll check after the tests

```qml
    function isValidAddress(inputValue) {
        return (inputValue !== "0x" && /^0x[a-fA-F0-9]{40}$/.test(inputValue)) || /^[a-zA-Z0-9.-]+\.eth$/.test(inputValue);
    }
```

**Comment by kounkou** [comment]
Sure, makes sense, thank you for clarifying. 
So as discussed offline, I am just updating the validation function and performing some tests for now.

This modification assumes that an ENS name is of the form `*.eth` where is * is any strings.
I wonder if we would like to add a limit to the length though. I'll check after the tests

```qml
    function isValidAddress(inputValue) {
        return (inputValue !== "0x" && /^0x[a-fA-F0-9]{40}$/.test(inputValue)) || /^[a-zA-Z0-9.-]+\.eth$/.test(inputValue);
    }
```

### Result

Updating the validation is straightforward, but backend might require some update.

```bash
DBG 2024-05-01 13:58:41.213-07:00 NewBE_callPrivateRPC                       topics="rpc" tid=1553990 file=core.nim:27 rpc_method=wallet_getSuggestedFees
DBG 2024-05-01 13:58:41.579-07:00 NewBE_callPrivateRPC                       topics="rpc" tid=1553990 file=core.nim:27 rpc_method=communitytokens_estimateMintTokens
ERR 2024-05-01 13:58:41.748-07:00 rpc response error                         topics="rpc" tid=1553990 file=core.nim:36 err="\nstatus-go error [methodName:communitytokens_estimateMintTokens, code:3, message:execution reverted: ERC20: mint to the zero address ]\n"
ERR 2024-05-01 13:58:41.748-07:00 error doing rpc request                    topics="rpc" tid=1553990 file=core.nim:40 methodName=communitytokens_estimateMintTokens exception="\nstatus-go error [methodName:communitytokens_estimateMintTokens, code:3, message:execution reverted: ERC20: mint to the zero address ]\n"
```

![Screenshot from 2024-05-01 13-58-42](https://github.com/status-im/status-desktop/assets/2589171/cfd7dbd0-4ad1-49e7-8966-fb68836bc0f1)

**Comment by jrainville** [comment]
Reopening as we will implement this later. I'll make this a feature insteam
